cmd/ignite/run/create.go

```diff
@@ -15,99 +15,144 @@ import (
 	"github.com/weaveworks/ignite/pkg/providers"
 )
 
+// NewCreateFlags returns an initialized CreateFlags instance.
 func NewCreateFlags() *CreateFlags {
-	return &CreateFlags{
-		VM: providers.Client.VMs().New(),
-	}
+	return &CreateFlags{}
 }
 
+// CreateFlags contains flag variables for create command.
 type CreateFlags struct {
 	PortMappings []string
 	CopyFiles    []string
 	// This is a placeholder value here for now.
 	// If it was set using flags, it will be copied over to
 	// the API type. TODO: When we later have internal types
 	// this can go away
-	SSH        api.SSH
-	ConfigFile string
-	VM         *api.VM
-	Labels     []string
+	SSH           api.SSH
+	ConfigFile    string
+	Labels        []string
+	VMName        string
+	VMCPUs        uint64
+	KernelCmdLine string
+	VMMemory      meta.Size
+	VMDiskSize    meta.Size
+	// Use string for KernelOCI instead of OCIImageRef because OCIImageRef can't
+	// be created without setting a valid value. Need zero valued variable to
+	// know if the flag was set or not.
+	KernelOCI string
+	VMStorage api.VMStorageSpec
```

Comment:
This replaced the VM object with individual elements to be able to know if a flag is set or not. When using a VM object, the VM properties are set to the defaults. Using that as flag variables makes it hard to know if a flag was set or empty. With this, these properties are set on the base VM configuration only when the value of these flag variables are non-zero.